Source/DafnyCore/Compilers/ExecutableBackend.cs

```diff
@@ -137,6 +139,14 @@ public abstract class ExecutableBackend : Plugins.IExecutableBackend {
     return true;
   }
 
+  public override void InstrumentCompiler(CompilerInstrumenter instrumenter, Program dafnyProgram) {
+    if (compiler == null) {
```

Comment:
Under what circumstances might the compiler be `null`?

Reply:
The LibraryBackend is the main reason :) ExecutableBackends don't HAVE to have a SinglePassCompiler in general.